### DISCUSSION TITLE
Open SSL

### DISCUSSION THREAD

**Discussion by dovantuan02** [paullouisageneau/libdatachannel#1230] [General]
hi everyone. I need help with error : src/impl/tls.hpp:71:10: fatal error: openssl/ssl.h: No such file or directory
![image](https://github.com/paullouisageneau/libdatachannel/assets/108563552/74ce2387-b24d-41d1-bdc2-292105abacfb)

**Comment by paullouisageneau** [comment]
libdatachannel depends on OpenSSL. It looks like you are cross-compiling, so you need OpenSSL built for your target architecture, and your toolchain should be able to find it. This is not a libdatachannel issue.

Additionnally, I would recommend using CMake instead of the simple Makefile for cross-compiling. Your SDK should provide a CMake toolchain that you need to specify with `-DCMAKE_TOOLCHAIN_FILE`.

**Comment by dovantuan02** [comment]
I have toolchain and openssl. I need build crosscompile with libdatachannel. Now, I don't know how to do that. Can't you help me with Makefile please ?

**Reply by paullouisageneau** [reply]
To cross-compile with CMake, refer to [BUILDING.md](https://github.com/paullouisageneau/libdatachannel/blob/master/BUILDING.md). The commands are basically:
```
$ cmake -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=/your/toolchain.cmake -DOPENSSL_ROOT_DIR=/your/openssl/dir/or/prefix
$ cd build
$ make
```
